Find a functionally equivalent code:com.android.internal.util.XmlUtils.readThisMapXml:COMMENT
Method Modifier: public      static      final       internal    
Comment:/**
 * Read a HashMap object from an XmlPullParser.  The XML data could
 * previously have been generated by writeMapXml().  The XmlPullParser
 * must be positioned <em>after</em> the tag that begins the map.
 *
 * @param parser The XmlPullParser from which to read the map data.
 * @param endTag Name of the tag that will end the map, usually "map".
 * @param name An array of one string, used to return the name attribute
 * of the map's tag.
 *
 * @return HashMap The newly generated map.
 *
 * @see #readMapXml
 */

@@ -1,22 +1,3 @@
 {
-    HashMap map = new HashMap();
-    int eventType = parser.getEventType();
-    do {
-        if (eventType == parser.START_TAG) {
-            Object val = readThisValueXml(parser, name);
-            if (name[0] != null) {
-                // System.out.println("Adding to map: " + name + " -> " + val);
-                map.put(name[0], val);
-            } else {
-                throw new XmlPullParserException("Map value without name attribute: " + parser.getName());
-            }
-        } else if (eventType == parser.END_TAG) {
-            if (parser.getName().equals(endTag)) {
-                return map;
-            }
-            throw new XmlPullParserException("Expected " + endTag + " end tag at: " + parser.getName());
-        }
-        eventType = parser.next();
-    } while (eventType != parser.END_DOCUMENT);
-    throw new XmlPullParserException("Document ended before " + endTag + " end tag");
+    return readThisMapXml(parser, endTag, name, null);
 }

